modules/promotion/src/PromotionUsageInterface.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+namespace Drupal\commerce_promotion;
+
+use Drupal\commerce_order\Entity\OrderInterface;
+use Drupal\commerce_promotion\Entity\CouponInterface;
+use Drupal\commerce_promotion\Entity\PromotionInterface;
+
+/**
+ * Provides a way to check the usage of a promotion or coupon.
+ */
+interface PromotionUsageInterface {
+
+  /**
+   * Gets the usage for a promotion.
+   *
+   * @param \Drupal\commerce_promotion\Entity\PromotionInterface $promotion
+   *   The promotion.
+   *
+   * @return int
+   *   The usage.
+   */
+  public function getPromotionUsage(PromotionInterface $promotion);
+
+  /**
+   * Add a promotion usage record.
+   *
+   * @param \Drupal\commerce_promotion\Entity\PromotionInterface $promotion
+   *   The promotion.
+   * @param \Drupal\commerce_order\Entity\OrderInterface $order
+   *   The order.
+   */
+  public function addPromotionUsage(PromotionInterface $promotion, OrderInterface $order);
+
+  /**
+   * Gets the usage for a coupon.
+   *
+   * @param \Drupal\commerce_promotion\Entity\CouponInterface $coupon
+   *   The coupon.
+   *
+   * @return int
+   *   The usage.
+   */
+  public function getCouponUsage(CouponInterface $coupon);
```

Comment:
Same here, getUsage($promotion, $coupon = NULL), and some documentation to clarify?